### DISCUSSION TITLE
AccessKit Alt Text: option to hide "Image" alt text

### DISCUSSION THREAD

**Discussion by marcustyphoon** [AprilSylph/XKit-Rewritten#390] [Suggestion Box]
While I think this option should probably be disabled by default to raise awareness of the fact that most images do not have any alt text (which becomes extremely noticeable when ~every image contains an "Image" label), it would be nice to have an option to disable this, and thus only display alt text on images which have alt text that does not match the default.

This should be doable, as the default alt text appears to always be "Image" translated into the current locale (`window.tumblr.languageData` should work for this).

**Comment by AprilSylph** [comment]
If it's always translated, then I agree. The way AccessKit is set up would make this difficult, but probably worth doing.

**Reply by AprilSylph** [reply]
I'm so sure that when I reviewed #194 I tested if the default alt text was submitted on upload and it was, but either I'm wrong about that or that was changed. Either way, it means this is feasible now.

**Reply by AprilSylph** [reply]
Implemented in 62dae6038aee05cac721464d18b5ce95b9239500.

**Reply by marcustyphoon** [reply]
It was; you suggested removing the feature at the time because of the localization thing and the lack of toggle, I believe :D

**Reply by marcustyphoon** [reply]
Wait, ignore that; I misinterpreted your message!

**Reply by AprilSylph** [reply]
It seems that quite a few image blocks do indeed have the alt text "image" on the API level, so Tumblr's frontend doesn't try to fill in for it and ergo doesn't use a translated string. This doesn't happen anymore (tested both legacy photos and legacy inline images, to be sure) but it does mean mileage may vary for this option depending on the images on the user's dashboard.

I don't plan to do anything about it - I think it's fine for the feature to display all API-given alt text - but it's sure interesting, and not something to leave unsaid!

**Reply by AprilSylph** [reply]
It appears that a noticeable amount of even recent image uploads have "image" set explicitly as their alt text. At the time of my previous comment, every "image" image was a year or more old... but now I'm convinced that either the mobile app(s) are still doing this, or people are using old app versions which do this. Either way, I've excluded them from counting as image descriptions in v0.14.1.

**Reply by marcustyphoon** [reply]
Interesting! I hadn't seen any on my dash, but there do indeed appear to be some in a few tags I checked. I changed the UI language and looked at a few foreign-language tags and the 0.14.1 fix seems  to work fine.

**Reply by AprilSylph** [reply]
I just collected a ton of examples for Tumblr support, and all of them were made with the legacy editor (`shouldOpenInLegacy` is `true`)... I'm convinced now that it's a legacy to NPF conversion bug.
